bundles/org.openhab.binding.goveelan/src/main/java/org/openhab/binding/goveelan/internal/GoveeLanHandler.java

```diff
@@ -0,0 +1,365 @@
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.util.ColorUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link GoveeLanHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * Any device has its own job that triggers a refresh of retrieving the external state from the device.
+ * However, there must be only one job that listens for all devices in a singleton thread because
+ * all devices send their udp packet response to the same port on openHAB. Based on the sender IP address
+ * of the device we can detect to which thing the status answer needs to be assigned to and updated.
+ *
+ * <ul>
+ * <li>The job per thing that triggers a new update is called <i>triggerStatusJob</i>. There are as many instances
+ * as things.</li>
+ * <li>The job that receives the answers and applies that to the respective thing is called <i>refreshStatusJob</i> and
+ * there is only one for all instances. It may be stopped and restarted by the DiscoveryService (see below).</li>
+ * </ul>
+ *
+ * The other topic that needs to be managed is that device discovery responses are also sent to openHAB at the same port
+ * like status updates. Therefore, when scanning new devices that job that listens to status devices must
+ * be stopped while scanning new devices. Otherwise, the status job will receive the scan discover UDB packages.
+ *
+ * @author Stefan Höhn - Initial contribution
+ */
+@NonNullByDefault
+public class GoveeLanHandler extends BaseThingHandler {
+
+    /*
+     * Messages to be sent to the Govee Devices
+     */
+    private static final String LIGHT_OFF = "{\"msg\": {\"cmd\": \"turn\", \"data\": {\"value\": \"0\"}}}";
+    // turning on via cmd-turn and value = 1 doesn't work, so let's use the brightness command
+    private static final String LIGHT_ON = """
+            {"msg": {"cmd": "brightness", "data": {"value": "100"}}}
+            """;
+    private static final String LIGHT_COLOR = """
+            {
+              "msg" : {
+                "cmd":"colorwc",
+                "data": {
+                  "color" : {
+                    "r" : %d,
+                    "g" : %d,
+                    "b":  %d
+                  },
+                  "colorTemInKelvin" : %d
+                }
+              }
+            }
+            """;
+    private static final String LIGHT_BRIGHTNESS = """
+            {
+              "msg":{
+                "cmd":"brightness",
+                "data":{
+                  "value": %d
+                }
+              }
+            }
+            """;
+
+    private static final String QUERY_STATUS = """
+            {
+              "msg":{
+                "cmd":"devStatus",
+                "data":{
+                }
+              }
+            }
+            """;
```

Comment:
I would recommend writing one or more DTO classes for serializing this into JSON. This will ensure correctness (avoiding syntax errors, for example) and increase readability.

Reply:
Added DTOs and tests for it.